gstreamer-pbutils/src/auto/functions.rs

```diff
@@ -0,0 +1,373 @@
+// This file was generated by gir (d933f9a+) from gir-files (???)
+// DO NOT EDIT
+
+use EncodingTarget;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use gst;
+use std::mem;
+
+
+pub fn codec_utils_aac_caps_set_level_and_profile(caps: &gst::Caps, audio_config: &[u8]) -> bool {
+    assert_initialized_main_thread!();
+    let len = audio_config.len() as u32;
+    unsafe {
+        from_glib(ffi::gst_codec_utils_aac_caps_set_level_and_profile(caps.to_glib_none().0, audio_config.to_glib_none().0, len))
+    }
+}
+
+pub fn codec_utils_aac_get_channels(audio_config: &[u8]) -> u32 {
+    assert_initialized_main_thread!();
+    let len = audio_config.len() as u32;
+    unsafe {
+        ffi::gst_codec_utils_aac_get_channels(audio_config.to_glib_none().0, len)
+    }
+}
+
+pub fn codec_utils_aac_get_index_from_sample_rate(rate: u32) -> i32 {
+    assert_initialized_main_thread!();
+    unsafe {
+        ffi::gst_codec_utils_aac_get_index_from_sample_rate(rate)
+    }
+}
+
+pub fn codec_utils_aac_get_level(audio_config: &[u8]) -> Option<String> {
+    assert_initialized_main_thread!();
+    let len = audio_config.len() as u32;
+    unsafe {
+        from_glib_none(ffi::gst_codec_utils_aac_get_level(audio_config.to_glib_none().0, len))
+    }
+}
+
+pub fn codec_utils_aac_get_profile(audio_config: &[u8]) -> Option<String> {
+    assert_initialized_main_thread!();
+    let len = audio_config.len() as u32;
+    unsafe {
+        from_glib_none(ffi::gst_codec_utils_aac_get_profile(audio_config.to_glib_none().0, len))
+    }
+}
+
+pub fn codec_utils_aac_get_sample_rate(audio_config: &[u8]) -> u32 {
+    assert_initialized_main_thread!();
+    let len = audio_config.len() as u32;
+    unsafe {
+        ffi::gst_codec_utils_aac_get_sample_rate(audio_config.to_glib_none().0, len)
+    }
+}
+
+pub fn codec_utils_aac_get_sample_rate_from_index(sr_idx: u32) -> u32 {
+    assert_initialized_main_thread!();
+    unsafe {
+        ffi::gst_codec_utils_aac_get_sample_rate_from_index(sr_idx)
+    }
+}
+
+pub fn codec_utils_h264_caps_set_level_and_profile(caps: &gst::Caps, sps: &[u8]) -> bool {
```

Comment:
This is changing the caps, it would need to take a &mut CapsRef. Same in the others

Reply:
Manually implement unfortunately, gir does not know about the COW writability of mini objects.